spec/services/yml_loader_spec.rb

```diff
@@ -0,0 +1,139 @@
+require 'rails_helper'
+
+RSpec.describe YmlLoader do
+  let(:degree_id) { 'test_degree' }
+  let(:another_degree_id) { 'another_test_degree' }
+
+  before do
+    allow(Rails.configuration).to receive(:degrees).and_return([
+      {
+        bedelias_name: 'TEST DEGREE',
+        id: degree_id,
+        current_plan: '2025',
+        include_inco_subjects: true
+      },
+      {
+        bedelias_name: 'ANOTHER TEST DEGREE',
+        id: another_degree_id,
+        current_plan: '1815',
+        include_inco_subjects: true
+      }
+    ])
+  end
+
+  describe '.load' do
+    context 'yamls are correct' do
+      let(:base_dir) { Rails.root.join("spec/support/mock_yamls/success") }
+
+      it 'loads' do
+        # Degrees
+        expect { described_class.load(base_dir:) }.to change(Degree, :count).by(2)
+        degree = Degree.find(degree_id)
+        expect(degree).to be_present
+        expect(degree.current_plan).to eq('2025')
+        expect(degree.include_inco_subjects).to be true
+
+        # Subject Groups
+        expect(degree.subject_groups.count).to eq(1)
+        subject_group = degree.subject_groups.find_by(code: '2003')
+        expect(subject_group).to be_present
+        expect(subject_group.name).to eq('Test Group')
+        expect(subject_group.credits_needed).to eq(70)
+
+        # Subjects
+        expect(degree.subjects.count).to eq(2)
+        subject1 = Subject.find_by(code: '101')
```

Comment:
por las dudas habria que hacer asi no? Pueden haber dos con el mismo codigo pero de degrees distintos
```suggestion
        subject1 = degree.subjects.find_by(code: '101')
```